lib/puppetserver/ca/action/list.rb

```diff
@@ -79,41 +88,62 @@ def run(input)
           requested, signed, revoked = separate_certs(all_certs)
           missing = certnames - all_certs.map { |cert| cert['name'] }
 
-          (all || certnames.any?) \
-            ? output_certs_by_state(requested, signed, revoked, missing)
-            : output_certs_by_state(requested)
+          (all || certnames.any?)\
+            ? output_certs_by_state(all, output_format, requested, signed, revoked, missing)
+            : output_certs_by_state(all, output_format, requested)
 
           return missing.any? \
             ? 1
             : 0
         end
 
-        def output_certs_by_state(requested, signed = [], revoked = [], missing = [])
+        def output_certs_by_state(all, output_format, requested, signed = [], revoked = [], missing = [])
```

Comment:
sooo I kinda think we should include all keys in the JSON output, even if some of the values are empty lists..
I feel like that will require less error-handling from consumers, which seems nice?